Source/Evolve/WarpXEvolve.cpp

```diff
@@ -523,6 +514,17 @@ void WarpX::HandleParticlesAtBoundaries (int step, amrex::Real cur_time, int num
         }
     }
 
+    mypc->ApplyBoundaryConditions();
```

Comment:
I'm not sure that this is correct. If I understand it, `Redistribute` not only sends particles to their appropriate processor, but it also deletes any particles outside of the domain. You can see this in the routine `locateParticle` in amrex/Src/Particle/AMReX_ParticleContainerI.H. So, `ApplyBoundaryConditions` would have to be called before `Redistribute` to handle the reflecting and thermal boundary conditions.